pkg/storage/stores/indexshipper/uploads/table.go

```diff
@@ -0,0 +1,137 @@
+package uploads
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log"
+
+	"github.com/grafana/loki/pkg/storage/stores/indexshipper/index"
+	"github.com/grafana/loki/pkg/storage/stores/shipper/storage"
+	util_log "github.com/grafana/loki/pkg/util/log"
+)
+
+const (
+	// a temp file is created during uploads with name of the db + tempFileSuffix
+	tempFileSuffix = ".temp"
+)
+
+type Table interface {
+	Name() string
+	AddIndex(userID string, idx index.Index) error
+	ForEach(userID string, callback index.ForEachIndexCallback) error
+	Upload(ctx context.Context) error
+	Cleanup(indexRetainPeriod time.Duration) error
+	Stop()
+}
+
+// table is a collection of multiple dbs created for a same table by the ingester.
```

Comment:
Can we add a bit more detail here, as someone new to the index and shipper part of the code base I still don't know what a table is in this context. Is the name a side effect of having used boltdb up until this point? does it refer to all indexes for a given configuration, such as by time range or tenant?

Reply:
`table` holds index for logs within a specific interval, `24h` in case of `boltdb-shipper`. I will add more comments.